### PR TITLE
Fix LZO decompression

### PR DESCRIPTION
**Pull request by majetideepak**
resolves https://github.com/facebookincubator/velox/issues/9618

The output is verified via Presto.

### CODE REVIEW

**Comment by majetideepak** [comment]
@Yuhta Can you take a look? Thanks!

**Review by yingsu00 — APPROVED** [review]
@majetideepak Thanks for fixing this! The bug also indicates insufficient tests to cover different compression schemes in existing E2EFilterTest. https://github.com/facebookincubator/velox/issues/8103. We will need to make the tests to loop over different compression schemes in future PRs.

**Comment by majetideepak** [comment]
> The bug also indicates insufficient tests to cover different compression schemes in existing E2EFilterTest

@yingsu00 We don't currently have support for lzo compression. We should add that support as well.

**Comment by majetideepak** [comment]
@pedroerp can you please help review this? Thanks.

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp [hunk 1]
```diff
@@ -1181,3 +1181,37 @@ TEST_F(ParquetReaderTest, testV2PageWithZeroMaxDefRep) {
   assertReadWithReaderAndExpected(
       outputRowType, *rowReader, expected, *leafPool_);
 }
+
+TEST_F(ParquetReaderTest, testLzoDataPage) {
+  const std::string sample(getExampleFilePath("lzo.parquet"));
+
+  facebook::velox::dwio::common::ReaderOptions readerOptions{leafPool_.get()};
```
Comment (pedroerp) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:1188]: nit: we should add tests to a "facebook::velox::dwio::test" sort of namespace, so you could just do "common::ReaderOptions" here

**Review by pedroerp — APPROVED** [review]
Thank you!

We should at some point have an actual fuzzer suite to cross test all sorts of file formats and compression schemes.

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**